File: 卡码/代码随想录/1/01-数组/Go/01-二分/binarySearch.go
//704. 二分查找
package main

import "fmt"

func search(nums []int, target int) int {
	left,right:=0,len(nums)-1
	for left<=right {
		middle:=(left+right)/2
		if nums[middle]==target {
			return middle
		}else if target<nums[middle] {
			right=middle-1
		}else{
			left=middle+1
		}
	}
	return -1
}

func main(){
	nums := []int{-1,0,3,5,9,12}
	fmt.Println(search(nums,9))
	fmt.Println(search(nums,2))
	fmt.Println(search(nums,13))
}

File: 卡码/代码随想录/1/01-数组/C++/01-二分/binarySearch.cpp
//704. 二分查找
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
  int search(vector<int>& nums, int target) {
		int left=0,right=nums.size()-1;
		while(left<=right){
			int middle=(left+right)/2;
			if(nums[middle]==target) return middle;
			else if(target<nums[middle]) right=middle-1;
			else left=middle+1;
		}
		return -1;
  }
};

int main(){
	Solution solution=new Solution();
	vector<int> nums(6);
	nums.push_back(-1);
	nums.push_back(0);
	nums.push_back(3);
	nums.push_back(5);
	nums.push_back(9);
	nums.push_back(12);
	cout<<solution->search(nums,9);
	cout<<solution->search(nums,2);
	cout<<solution->search(nums,13);
	return 0;
}